include/odom_predictor/imu_integrator.h

```diff
@@ -0,0 +1,105 @@
+#ifndef ODOM_PREDICTOR_IMU_INTEGRATOR_H
+#define ODOM_PREDICTOR_IMU_INTEGRATOR_H
+
+#include <Eigen/Dense>
+#include <kindr/minimal/quat-transformation.h>
+#include <boost/shared_ptr.hpp>
+
+typedef kindr::minimal::QuatTransformation Transformation;
+typedef kindr::minimal::RotationQuaternion Rotation;
+typedef Transformation::Vector3 Vector3;
+
+class ImuIntegrator {
+
+ public:
+  virtual void resetState(const Transformation& transform = Transformation(),
+                          const Vector3& linear_velocity = Vector3::Zero(),
+                          const Vector3& angular_velocity = Vector3::Zero()) = 0;
+
+  virtual void resetBias(const Vector3& imu_angular_velocity_bias = Vector3::Zero(),
+                         const Vector3& imu_linear_acceleration_bias = Vector3::Zero()) = 0;
+
+  virtual void addMeasurement(const double dt,
+                              const Vector3& imu_linear_acceleration,
+                              const Vector3& imu_angular_velocity) = 0;
+
+  virtual void getState(Transformation* transform,
+                        Vector3* linear_velocity,
+                        Vector3* angular_velocity) = 0;
+
+};
+
+class ZecImuIntegrator : ImuIntegrator {
```

Comment:
Hahahahahahahaha this is amazing
@ZacharyTaylor your legacy lives on!!!

Reply:
had to name it somehow :sheep: